tests/src/JIT/HardwareIntrinsics/X86/Shared/BooleanCmpOpTest.template

```diff
@@ -0,0 +1,196 @@
+        private {3}<{2}> _fld2;
+
+        private BooleanComparisonOpTest__DataTable<{2}> _dataTable;
+
+        static BooleanComparisonOpTest__{1}{2}()
+        {{
+            var random = new Random();
+
+            for (var i = 0; i < ElementCount; i++) {{ _data1[i] = {5}; _data2[i] = {5}; }}
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<{3}<{2}>, byte>(ref _clsVar1), ref Unsafe.As<{2}, byte>(ref _data2[0]), VectorSize);
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<{3}<{2}>, byte>(ref _clsVar2), ref Unsafe.As<{2}, byte>(ref _data1[0]), VectorSize);
+        }}
+
+        public BooleanComparisonOpTest__{1}{2}()
+        {{
+            Succeeded = true;
+
+            var random = new Random();
+
+            for (var i = 0; i < ElementCount; i++) {{ _data1[i] = {5}; _data2[i] = {5}; }}
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<{3}<{2}>, byte>(ref _fld1), ref Unsafe.As<{2}, byte>(ref _data1[0]), VectorSize);
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<{3}<{2}>, byte>(ref _fld2), ref Unsafe.As<{2}, byte>(ref _data2[0]), VectorSize);
+
+            for (var i = 0; i < ElementCount; i++) {{ _data1[i] = {5}; _data2[i] = {5}; }}
+            _dataTable = new BooleanComparisonOpTest__DataTable<{2}>(_data1, _data2);
+        }}
+
+        public bool IsSupported => {0}.IsSupported;
+
+        public bool Succeeded {{ get; set; }}
+
+        public void RunBasicScenario()
+        {{
+            var result = {0}.{1}(
+                Unsafe.Read<{3}<{2}>>(_dataTable.inArray1Ptr),
+                Unsafe.Read<{3}<{2}>>(_dataTable.inArray2Ptr)
+            );
+
+            ValidateResult(_dataTable.inArray1, _dataTable.inArray2, result);
+        }}
+
+        public void RunReflectionScenario()
+        {{
+            var result = typeof({0}).GetMethod(nameof({0}.{1}), new Type[] {{ typeof({3}<{2}>), typeof({3}<{2}>) }})
+                                     .Invoke(null, new object[] {{
+                                        Unsafe.Read<{3}<{2}>>(_dataTable.inArray1Ptr),
+                                        Unsafe.Read<{3}<{2}>>(_dataTable.inArray2Ptr)
+                                     }});
+
+            ValidateResult(_dataTable.inArray1, _dataTable.inArray2, (bool)(result));
+        }}
+
+        public void RunClsVarScenario()
+        {{
+            var result = {0}.{1}(
+                _clsVar1,
+                _clsVar2
+            );
+
+            ValidateResult(_clsVar1, _clsVar2, result);
+        }}
+
+        public void RunLclVarScenario()
+        {{
+            var left = Unsafe.Read<{3}<{2}>>(_dataTable.inArray1Ptr);
+            var right = Unsafe.Read<{3}<{2}>>(_dataTable.inArray2Ptr);
+            var result = {0}.{1}(left, right);
+
+            ValidateResult(left, right, result);
+        }}
+
+        public void RunLclFldScenario()
+        {{
+            var test = new BooleanComparisonOpTest__{1}{2}();
+            var result = {0}.{1}(test._fld1, test._fld2);
```

Comment:
Hmm.. Looks like we need another template for Avx comparisons that take two vectors and an imm value.

Reply:
Yes, I think we will end up having at least 5-10 templates to support all tests, but this is much more manageable than 500-1000 individual tests that need handling 😉 

I think it is even fine if the templates are "nearly" identical in some cases, as they are still few in number and it makes the overall logic to maintain them simpler.